src/params.rs

```diff
@@ -738,10 +740,12 @@ impl<'a> UnitaryGenerator for ParamManager {
         for e in &seq {
             permut.push(*e as u32);
         }
-        rand::thread_rng().shuffle(&mut permut);
         let mut permut_inv: Vec<u32> = permut.clone();
-        for i in 0..permut.len(){
-            permut_inv[permut[i] as usize] = i as u32;
+        if is_permut_const == false {
```

Comment:
Would this be useful anywhere besides the test? If so why can't you just set this permut vector within the tests itself to be fixed? I..e maybe during the unitary layer's creation in layer_builder?